actionview/lib/action_view/template/handlers/erb.rb

```diff
@@ -1,68 +1,65 @@
-require "erubis"
+require "erubi"
 
 module ActionView
   class Template
     module Handlers
-      class Erubis < ::Erubis::Eruby
-        def add_preamble(src)
+      class Erubi < ::Erubi::Engine
+        def initialize(input, properties = {})
           @newline_pending = 0
-          src << "@output_buffer = output_buffer || ActionView::OutputBuffer.new;"
+
+          properties = Hash[properties]
```

Comment:
Because otherwise you mutate an input hash that is passed in.  Mutation is faster, but it's generally considered a bad idea to mutate something you don't own, and it could lead to undesired behavior if the constructor is called directly.